src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -177,79 +177,58 @@ public void DirectoryInfo_Create_NullDirectoryInfo()
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            Assert.Throws<ArgumentNullException>("directoryInfo", () => DirectoryInfo_Create_Framework(info, security));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NullDirectorySecurity()
         {
             DirectoryInfo info = new DirectoryInfo("path");
-
-            Assert.Throws<ArgumentNullException>("directorySecurity", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, null);
-                }
-                else
-                {
-                    info.Create(null);
-                }
-            });
+            Assert.Throws<ArgumentNullException>("directorySecurity", () => DirectoryInfo_Create_Framework(info, null));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            using var directory = new TempDirectory();
-            string path = Path.Combine(directory.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
+            var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "ParentDoesNotExist");
+            directory.Dispose(); // Delete parent folder
+
             DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            DirectorySecurity security = GetDirectorySecurity(FileSystemRights.FullControl);
+            DirectoryInfo_Create_Framework(info, security);
+        }
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
+        private void DirectoryInfo_Create_Framework(DirectoryInfo info, DirectorySecurity security)
+        {
+            if (PlatformDetection.IsNetFramework)
             {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+                FileSystemAclExtensions.Create(info, security);
+            }
+            else
+            {
+                info.Create(security);
+            }
         }
 
         [Fact]
-        public void DirectoryInfo_Create_DefaultDirectorySecurity()
+        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule()
         {
-            DirectorySecurity security = new DirectorySecurity();
-            Verify_DirectoryInfo_Create(security);
-        }
+            using var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "directory");
+            DirectoryInfo info = new DirectoryInfo(path);
 
-        [Theory]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
```

Comment:
Do we still have sufficient coverage of all these somewhere?

Reply:
There are other unit test files in this assembly that verify other `FileSystemRights` are correctly set.
Maybe I should bring back _some_ of the InlineData combinations. I noticed some combinations were either not accepted or would prevent me from deleting files. I think anyth rights involving `Read`, `Write` or `Execute` in any combination should probably be ok.

Reply:
So the only `FileSystemRights` I could bring back to the `InlineData` would be those that have a `Read` flag. Otherwise, the `DeleteDirectory` method we call when disposing will fail to enumerate the folders. Makes sense.

Reply:
I brought back all the ones I had deleted.